### PR TITLE
JWT workato claim

### PR DESCRIPTION
**Pull request by eeliang**
### Which issues does this fix?
Docs for https://github.com/workato/issues/issues/11109

### PR submission checklist
- [x] Checked all content for grammatical errors (using Grammarly, for example)
- [x] Added new files to `docs/vuepress/sidebar.js`
- [x] Check for broken links
```bash
npm run check
```

New ToC:
```
- Managing API Clients and Access Profiles
- - Access Token
- - Auth token
- - JWT token
- - JWT Workato claims
```

Created:
- auth-token.md
- jwt-token.md
- jwt-workato-claim.md

Modified:
- access-tokens.md (as a place holder. Please check content)
- api-client-mgmt.md
- api-security.md
- custom-domain.md
- /features/callable-recipes/new-call-trigger.md

### CODE REVIEW

**Comment by eeliang** [comment]
JWT specs: https://docs.google.com/document/d/10VUrQMBXp5Aj2B3RHLjdDIeiRIE5ksKlozGev2A5AT0/edit?ts=5e7304b1#heading=h.xa1y6yegyear
JWT sub priority: https://docs.google.com/document/d/1kYTvGfCnk8HVYr2EM2N9zqLdCAE1X34AX03o9MNNB-M/edit

File: docs/api-mgmt/auth-token.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+An Auth token is the simplest way to authenticate the API request with Workato.
+
+First, retrieve the API token from the access profile. Use the copy icon to copy the API token to your clipboard.
+
+![Auth Token for an Access profile](~@img/api-mgmt/auth-token.png)
+*Auth Token for an Access profile*
+
+Next, configure a header with your API token value. For example:
+
+```
```
Comment (eeshansim) [inline docs/api-mgmt/auth-token.md:17]: `bash` type

File: docs/api-mgmt/access-tokens.md [hunk 2]
```diff
@@ -2,114 +2,22 @@
...
-Next, select a collection in the next screen. To test an API endpoint within a collection, click on the toggle for **Authentication method and token** and select JSON web token (JWT). Paste the JWT token here and set the token.
-
-![Select JWT](~@img/api-mgmt/set-jwt-token.png)
-*Setting the JWT token*
-
-After setting the token, choose on one of the endpoints and provide valid values for the required parameters. Click on **Test** at the bottom of the page to test the API endpoint. The test should return a `200` response code and the expected response body.
-
-![Select JWT](~@img/api-mgmt/test-success.png)
-*200 response code and response body*
+Access tokens are string fields that identify the client of a Callable Recipe. The token value is a secret that is shared between a client and the Workato server. A token is passed to the API in an authorization header. The header must have a valid value for the call to succeed.
```
Comment (eeshansim) [inline docs/api-mgmt/access-tokens.md:8]: Access tokens are ~string fields~ strings that...

File: docs/api-mgmt/api-client-mgmt.md [hunk 3]
```diff
@@ -4,43 +4,84 @@ date: 2018-04-08 10:43:00 Z
...
-| Authentication method<br>*(Required)* | This can be an Auth Token or a JSON Web Token (JWT). |
+Note the **API key** field. A unique API token is generated for each client. This token is a long string of characters. It needs to be supplied to the client so that the client can connect to the API. Treat this API key as confidential information: it should be known only to the API owner and the client.
 
-If "JSON Web Token (JWT)" is selected there are two additional fields available: **signing method**, and **secret or key**. See [Access Tokens](/api-mgmt/access-tokens.md) for details.
+An API token can be revoked, and a new one issued, by clicking on the **Refresh** button next to the token.
 
-Once the access profile fields have been filled in, click on **Add profile** to create the client and save that client's associated profile. After this step, the screen will look something like this:
+A client can be **Disabled** or **Enabled**. A disabled client cannot call any APIs. Moving the slider right will switch the client's status to **Enabled**, after which he/she will be allowed to make API calls.
```
Comment (eeshansim) [inline docs/api-mgmt/api-client-mgmt.md:37]: ...after which API calls will be accepted.

File: docs/api-mgmt/jwt-token.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
+---
+title: JWT Token
+date: 2020-04-18 12:26:00 Z
+---
+
+# JSON Web Token
+
+For additional security, API consumers can now make use of JSON Web Tokens (JWT). This is a standard [RFC 7159](https://tools.ietf.org/html/rfc7519) for web authentication. JWT tokens are signed using a secret or key selected by the manager of the [access profile](/api-mgmt/api-client-mgmt.md#access-profile).
```
Comment (eeshansim) [inline docs/api-mgmt/jwt-token.md:8]: ...API consumers can ~now~ make use of...

File: docs/api-mgmt/jwt-token.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+JWT tokens are commonly used with identity providers (for example Okta, OneLogin, Auth0) who authenticate users and provide verified access to multiple applications. Also known as Single-Sign On, identity providers streamline access to applications through a central authentication mechanism.
```
Comment (eeshansim) [inline docs/api-mgmt/jwt-token.md:10]: JWT tokens are ~commonly~ used ~with~ by identity providers (for example Okta, OneLogin, Auth0) ~who~ that authenticate users and provide verified access to ~multiple~ business applications...

File: docs/api-mgmt/jwt-token.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+```json
+{
+  "sub": "d4aee74e131926682043395edecaf377765fae48e56e232cb295af475b314545",
+  "name": "John Doe"
+}
+```
+
+Here `sub` is a claim that denotes the principal subject (or API requester). The `sub` value is the API token that is obtainable from the [Access Profile screen](/api-mgmt/api-client-mgmt.md#access-profile) in Workato. Lastly, `name` is the name of the requester.
+
+Workato will inspect the `sub` claim to look for a valid API token before accepting the API request. If the `sub` claim is restricted by the identity provider and you are unable to provide the API token in this claim, you can configure custom Workato claims to store the API token. See [here](/api-mgmt/jwt-workato-claim.md).
```
Comment (eeshansim) [inline docs/api-mgmt/jwt-token.md:81]: If the `sub` claim is restricted by the identity provider and you are unable to provide the API token in this claim, you can configure custom ~Workato claims~ **Workato claims** to store the API token. ~See~ Learn more [here](/api-mgmt/jwt-workato-claim.md).

File: docs/api-mgmt/jwt-token.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+The JWT token is a signed representation of the JSON structure. You can generate a JWT token using the tools at [JWT.IO](https://jwt.io/).
```
Comment (eeshansim) [inline docs/api-mgmt/jwt-token.md:83]: why JWY.IO instead of jwt.io?

File: docs/api-mgmt/jwt-token.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+With JWT, the API requester is responsible for generating and packaging a token in the correct format. An online tool is available at [JWT.IO](https://jwt.io/) to facilitate this. The JSON format text mentioned above should be pasted or typed into the "payload" field on the "Decoded" side of the tool. In addition, they will paste in the private key (RSA) or secret string (HMAC) in the "Verify Signature" section. The signed and encoded key then appears on the right-hand side in the "Encoded" section.
```
Comment (eeshansim) [inline docs/api-mgmt/jwt-token.md:85]: ...The signed and encoded key then appears on the ~right~ left-hand side in the "Encoded" section.

File: docs/api-mgmt/jwt-workato-claim.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
+---
+title: JWT Token
+date: 2020-04-18 12:26:00 Z
+---
+
+# JWT Workato claim
+
+Identity providers streamline the process of maintaining verified access to multiple applications. The end user only needs to authenticate with the identity provider. Subsequently, the end user can access multiple applications and services without needing to remember additional sets of credentials. For example, the identity provider will issue JWT tokens that allow the end user to make authenticated requests with Workato API platform.
+
+![Identity provider issues JWT to the end user, who uses it to obtain verified access to Workato API platform](~@img/api-mgmt/jwt-flow.png)
```
Comment (eeshansim) [inline docs/api-mgmt/jwt-workato-claim.md:10]: Use a nicer drawing

File: docs/api-mgmt/jwt-workato-claim.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+| `https://www.workato.com/sub ` | This is a namespace claim. As it uses unique names, this claim is unlikely to be restricted by the identity providers. |
+| `workato_sub` | This is a payload claim. Workato will inspect this claim if the above claims are empty. |
+| `kid` | This is a header claim. Workato will inspect this claim if the above claims are empty. |
+| `sub` | This represents the subject of the JWT. Some identity providers reserve this JWT claim and thus Workato API token cannot be used here. Workato will inspect this claim if the above claims are empty. |
+
+## How to configure Workato claim
+
+Let’s go through an example of configuring a workato claim for Auth0. This identity provider restricts the `kid` and `sub` claims. Hence, we will be configuring a namespace claim to hold our API token.
+
+![Configure JWT in Auth0](~@img/api-mgmt/auth0-configure-jwt-flow.png)
```
Comment (eeshansim) [inline docs/api-mgmt/jwt-workato-claim.md:28]: Use a nicer drawing

File: docs/api-mgmt/jwt-workato-claim.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+![Identity provider issues JWT to the end user, who uses it to obtain verified access to Workato API platform](~@img/api-mgmt/jwt-flow.png)
+*Identity provider issues JWT to the end user, who uses it to obtain verified access to Workato API platform*
+
+When Workato receives an incoming request, the JWT token is checked to see if it contains a valid API Token. Workato does this to authenticate if the request is coming from a valid access profile. If no valid token is found, the API request will return a `403 forbidden` error.
+
+Workato will inspect the following JWT claims in sequential order. Workato identifies the **first claim** that is not empty and compares the claim value with an internal list of verified access profiles. If the token is not verified, the API request will return a `403 forbidden` error. Otherwise, if a valid API token is found, the API request will be successful.
+
+| JWT claims | Description |
```
Comment (eeshansim) [inline docs/api-mgmt/jwt-workato-claim.md:17]: This table can be clearer. Add a column called *Priority* - 1st, 2nd...

File: docs/api-mgmt/jwt-workato-claim.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+---
+
+# JWT Workato claim
+
+Identity providers streamline the process of maintaining verified access to multiple applications. The end user only needs to authenticate with the identity provider. Subsequently, the end user can access multiple applications and services without needing to remember additional sets of credentials. For example, the identity provider will issue JWT tokens that allow the end user to make authenticated requests with Workato API platform.
+
+![Identity provider issues JWT to the end user, who uses it to obtain verified access to Workato API platform](~@img/api-mgmt/jwt-flow.png)
+*Identity provider issues JWT to the end user, who uses it to obtain verified access to Workato API platform*
+
+When Workato receives an incoming request, the JWT token is checked to see if it contains a valid API Token. Workato does this to authenticate if the request is coming from a valid access profile. If no valid token is found, the API request will return a `403 forbidden` error.
```
Comment (eeshansim) [inline docs/api-mgmt/jwt-workato-claim.md:13]: When Workato receives an incoming request, the JWT token is checked to see if it contains a valid API Token. ~Workato does this to authenticate if~ This is done to determine that the request is coming from a valid access profile.

File: docs/api-mgmt/jwt-workato-claim.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+
+Workato will inspect the following JWT claims in sequential order. Workato identifies the **first claim** that is not empty and compares the claim value with an internal list of verified access profiles. If the token is not verified, the API request will return a `403 forbidden` error. Otherwise, if a valid API token is found, the API request will be successful.
```
Comment (eeshansim) [inline docs/api-mgmt/jwt-workato-claim.md:15]: ...internal list of ~verified~ known access profiles...

File: docs/api-mgmt/jwt-workato-claim.md (hunk 11 above)
Comment (eeshansim) [reply docs/api-mgmt/jwt-workato-claim.md:17]: Add a column called *Part* - header, payload

File: docs/api-mgmt/jwt-workato-claim.md [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+| JWT claims | Description |
+| ---------- | ----------- |
+| `https://www.workato.com/sub ` | This is a namespace claim. As it uses unique names, this claim is unlikely to be restricted by the identity providers. |
+| `workato_sub` | This is a payload claim. Workato will inspect this claim if the above claims are empty. |
+| `kid` | This is a header claim. Workato will inspect this claim if the above claims are empty. |
+| `sub` | This represents the subject of the JWT. Some identity providers reserve this JWT claim and thus Workato API token cannot be used here. Workato will inspect this claim if the above claims are empty. |
+
+## How to configure Workato claim
+
+Let’s go through an example of configuring a workato claim for Auth0. This identity provider restricts the `kid` and `sub` claims. Hence, we will be configuring a namespace claim to hold our API token.
```
Comment (eeshansim) [inline docs/api-mgmt/jwt-workato-claim.md:26]: Link Auth0

Comment (eeshansim) [reply docs/api-mgmt/jwt-workato-claim.md:26]: Let’s go through an example of configuring a workato claim for ~Auth0~ [Auth0](https://auth0.com/). This identity provider restricts the `kid` and `sub` claims. Furthermore, Auth0 requires all custom claims to be namespace. Hence, we will be configuring a namespace claim to ~hold~ bear the API token.

File: docs/api-mgmt/jwt-workato-claim.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+| 6.    | In the Auth0 application, find the **Settings** tab > **Application Tokens** > **Advanced settings**<br>![Open advanced settings](~@img/api-mgmt/auth0-configure-advanced-settings.png)*Open advanced settings* |
+| 7.    | Find the **Certificate** tab > copy the **Signing certificate**. We will use this to configure the Workato access profile in the next stage.<br>We will turn to this page in the 3rd stage.<br>![Obtain RSA certificate](~@img/api-mgmt/auth-configure-copy-certificate.png)*Obtain RSA certicate* |
+
+## 2. Configure Workato access profile
+
+Next, configure an access profile in Workato and obtain the API token.
+
+| Steps | Description |
+| ----- | ----------- |
+| 1.    | Find **Tools** > **API platform** > slect a Client > **Create access profile**. See here for more information about [API clients](/api-mgmt/api-client-mgmt#api-clients).<br>![Open Workato API platform](~@img/api-mgmt/open-api-platform.png)*Open Workato API platform* |
```
Comment (eeshansim) [inline docs/api-mgmt/jwt-workato-claim.md:56]: ~slect~ select

File: docs/api-mgmt/jwt-workato-claim.md [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+| 2.    | Select **JSON Web Taken (JWT)** as the authentication method. Profide the **RSA certificate** from the previously stage. See here for more information about [configuring access profile](/api-mgmt/api-client-mgmt#create-new-access-profile).<br>![Provide RSA certificate](~@img/api-mgmt/auth0-configure-provide-certifcate.png)*Provide RSA certificate* |
```
Comment (eeshansim) [inline docs/api-mgmt/jwt-workato-claim.md:57]: ~See here for more information~ Learn more about configuring access profile [here](/api-mgmt/api-client-mgmt#create-new-access-profile).

File: docs/api-mgmt/jwt-token.md (hunk 7 above)
Comment (eeliang) [reply docs/api-mgmt/jwt-token.md:83]: The original article used "JWT.IO" in all caps.

**Pull request merged**